plugins/optimization-detective/class-od-url-metric.php

```diff
@@ -212,6 +218,128 @@ public static function get_json_schema(): array {
 			),
 			'additionalProperties' => false,
 		);
+
+		/**
+		 * Filters additional schema properties which should be allowed at the root of a URL metric.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param array<string, array{type: string}> $additional_properties Additional properties.
+		 */
+		$additional_properties = (array) apply_filters( 'od_url_metric_schema_root_additional_properties', array() );
+		if ( count( $additional_properties ) > 0 ) {
+			$schema['properties'] = self::extend_schema_with_optional_properties( $schema['properties'], $additional_properties, 'od_url_metric_schema_root_additional_properties' );
+		}
+
+		/**
+		 * Filters additional schema properties which should be allowed for an elements item in a URL metric.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param array<string, array{type: string}> $additional_properties Additional properties.
+		 */
+		$additional_properties = (array) apply_filters( 'od_url_metric_schema_element_item_additional_properties', array() );
+		if ( count( $additional_properties ) > 0 ) {
+			$schema['properties']['elements']['items']['properties'] = self::extend_schema_with_optional_properties(
+				$schema['properties']['elements']['items']['properties'],
+				$additional_properties,
+				'od_url_metric_schema_root_additional_properties'
+			);
+		}
+
+		return $schema;
+	}
+
+	/**
+	 * Extends a schema with additional optional properties.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array<string, mixed> $properties_schema     Properties schema to extend.
+	 * @param array<string, mixed> $additional_properties Additional properties.
+	 * @param string               $filter_name           Filter name used to extend.
+	 *
+	 * @return array<string, mixed> Extended schema.
+	 */
+	protected static function extend_schema_with_optional_properties( array $properties_schema, array $additional_properties, string $filter_name ): array {
+		$doing_it_wrong = static function ( string $message ) use ( $filter_name ): void {
+			_doing_it_wrong(
+				esc_html( "add_filter({$filter_name},...)" ),
+				esc_html( $message ),
+				'Optimization Detective n.e.x.t'
+			);
+		};
+		foreach ( $additional_properties as $property_key => $property_schema ) {
+			if ( ! is_array( $property_schema ) ) {
+				continue;
+			}
+			if ( isset( $properties_schema[ $property_key ] ) ) {
+				$doing_it_wrong(
+					sprintf(
+						/* translators: property name */
+						__( 'Disallowed override of existing schema property "%s".', 'optimization-detective' ),
+						$property_key
+					)
+				);
+				continue;
+			}
+			if ( ! isset( $property_schema['type'] ) || ! is_string( $property_schema['type'] ) ) {
```

Comment:
Actually, `type` can also [be an array](https://github.com/WordPress/wordpress-develop/blob/06f7c825842f0e55fea3b82f694b43e45d1c66df/src/wp-includes/rest-api.php#L2156-L2169).

```suggestion
			if ( ! isset( $property_schema['type'] ) ) {
```

Reply:
Fixed in 91fcd01e